components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/actuate/endpoint/CamelRoutesEndpoint.java

```diff
@@ -76,6 +78,19 @@ public void startRoute(String id) throws Exception {
         camelContext.getRouteController().startRoute(id);
     }
 
+    public String resetRoute(String id) throws Exception {
+        ManagedRouteMBean resetRoute = camelContext.getManagedRoute(id, ManagedRouteMBean.class);
+        if (resetRoute != null) {
+            resetRoute.reset(true);
+        } else {
+            String description = "The provided route id does not exists";
+            ResponseEntity<?> objectResponseEntity = ResponseEntity.status(HttpStatus.NO_CONTENT).body(description);
+            ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Please note that the current code returns a HTTP 204 status code.  As discussed on a previous PR, we should return HTTP 404 here.